Addressbook/src/main.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/ash
+
+source addressbook.sh
+
+echo "Welcome!"
+echo
+
+usage_menu(){
+	echo "Commands:"
+	echo "	1) Insert User"
+	echo "	2) Delete User"
+	echo "	3) Search User"
+	echo "	4) Edit User"
+	echo "	exit) Exit"
+}
+
+
+read_and_insert_user()
+{
+	read -p "User:" name
+	read -p "Email:" email
+	read -p "Phone:" phone
+
+	insert -u $name -e $email -p $phone
+}
+
+read_and_delete_user()
+{
+	echo "Delete user! If you want to skip the field for search type: enter"
+	read -p "User:" name
+	read -p "Email:" email
+	read -p "Phone:" phone
+	
+	comm=""
+	if [ "$name" != "" ]; then
+		comm="${comm} -u ${name}"
+	fi
+	if [ "$email" != "" ]; then
+		comm="${comm} -e ${email}"
+	fi
+	if [ "$phone" != "" ]; then
+		comm="${comm} -p ${phone}"
+	fi
+
+	delete $comm
+}
+
+read_and_search_user()
+{
+	echo "Search for user! If you want to skip the field for search type: enter"
+	read -p "User:" name
+	read -p "Email:" email
+	read -p "Phone:" phone
+	
+	comm=""
+	if [ "$name" != "" ]; then
+		comm="${comm} -u ${name}"
+	fi
+	if [ "$email" != "" ]; then
+		comm="${comm} -e ${email}"
+	fi
+	if [ "$phone" != "" ]; then
+		comm="${comm} -p ${phone}"
+	fi
+	echo "------------Result------------"
+	search $comm
```

Comment:
The database layer functions, such as `search()` could have an interface where it could handle something like `search user=${name} email=${email} ...`, and it could be consumed without `getopts`.
The concatenation of the params for `getopts` wouldn't need to repeated, either.